src/test/scripts/check_enterprise_merge.sh

```diff
@@ -1,55 +1,89 @@
+# the echo command. We do not want "set -x" to show the echo command. We only
+# want to see the actual message in the output of echo itself. This function is
+# a trick to do so. Read the StackOverflow post below to understand why this
+# works and what this works around.
+# Source: https://superuser.com/a/1141026/242593
+shopt -s expand_aliases
+alias echo='{ save_flags="$-"; set +x;} 2> /dev/null; echo_and_restore'
+echo_and_restore() {
+        builtin echo "$*"
+        #shellcheck disable=SC2154
+        case "$save_flags" in
+         (*x*)  set -x
+        esac
 }
 
-cd /tmp
-if [ ! -d citus-enterprise ]; then
-    git clone https://${GIT_USERNAME}:${GIT_TOKEN}@github.com/citusdata/citus-enterprise
-fi
+# List executed commands. This is done so debugging this script is easier when
+# it fails. It's explicitly done after git remote add so username and password
+# are not shown in CI output (even though it's also filtered out by CircleCI)
+set -x
 
-cd citus-enterprise
+# Clone current git repo (which should be community) to a temporary working
+# directory and go there
+GIT_DIR_ROOT="$(git rev-parse --show-toplevel)"
+TMP_GIT_DIR="$(mktemp --directory -t citus-merge-check.XXXXXXXXX)"
+git clone "$GIT_DIR_ROOT" "$TMP_GIT_DIR"
+cd "$TMP_GIT_DIR"
+
+# Fails in CI without this
 git config user.email "citus-bot@microsoft.com"
 git config user.name "citus bot"
 
-# reset repository into usable state if script ran before
-git fetch origin
-git reset --hard
-git checkout enterprise-master
-git reset --hard origin/enterprise-master
-
-branch_name="${CIRCLE_BRANCH}"
-
-# check if the branch on community exists on enterprise
-# the output will not be empty if it does
-if [ `git branch -r --list origin/$branch_name` ]
-then
-    try_merge enterprise-master origin/$branch_name
-else
-    # add community as a remote if not already added
-    set +e
-    if ! git ls-remote community > /dev/null; then
-        set -e
-        git remote add --no-tags community git@github.com:citusdata/citus.git
-    fi
-    set -e
-
-    # prevent pushes to community and update branch we care about
-    git remote set-url --push community no-pushing
-    git fetch community $branch_name
-
-    try_merge enterprise-master community/$branch_name
+# Disable "set -x" temporarily, because $ENTERPRISE_REMOTE contains passwords
+{ set +x ; } 2> /dev/null
+git remote add enterprise "$ENTERPRISE_REMOTE"
+set -x
+
+git remote set-url --push enterprise no-pushing
+
+# Fetch enterprise-master
+git fetch enterprise enterprise-master
+
+
+git checkout "enterprise/enterprise-master"
+
+if git merge --no-commit "origin/$PR_BRANCH"; then
+    echo "INFO: community PR branch could be merged into enterprise-master, so everything is good"
+    exit 0
 fi
+
+# undo partial merge
+git merge --abort
+
+if ! git fetch enterprise "$PR_BRANCH" ; then
+    echo "ERROR: enterprise/$PR_BRANCH was not found and community PR branch could not be merged into enterprise-master"
```

Comment:
Would it make sense to give a link to the readme whenever this script fails so that people know what to do?